vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/steps/TouchSteps.java

```diff
@@ -134,26 +135,41 @@ private Supplier<Rectangle> createSwipeArea()
         return () -> new Rectangle(new Point(0, 0), genericWebDriverManager.getSize());
     }
 
-    private void adjustVerticalPosition(WebElement element, Rectangle swipeArea, Duration swipeDuration)
+    private void adjustElementPosition(WebElement element, SwipeDirection direction, Rectangle swipeArea,
+            Duration swipeDuration)
     {
-        int swipeAreaSizeHeight = swipeArea.getHeight();
-        int swipeAreaCenterY = swipeAreaSizeHeight / 2;
-        int elementTopCoordinateY = element.getLocation().getY();
-
-        int bottomVisibilityIndent = (int) (VISIBILITY_BOTTOM_INDENT_COEFFICIENT * swipeAreaSizeHeight);
-        int visibilityY = swipeAreaSizeHeight - bottomVisibilityIndent;
-        if (elementTopCoordinateY > visibilityY)
+        int swipeAreaSize;
+        int elementCoordinate;
+        int bottomOrLeftVisibilityIndent;
+        int topOrRightVisibilityIndent;
+        if (SwipeDirection.UP == direction || SwipeDirection.DOWN == direction)
+        {
+            swipeAreaSize = swipeArea.getHeight();
+            elementCoordinate = element.getLocation().getY();
+            bottomOrLeftVisibilityIndent = (int) (VISIBILITY_BOTTOM_INDENT_COEFFICIENT * swipeAreaSize);
+            topOrRightVisibilityIndent = (int) (VISIBILITY_TOP_INDENT_COEFFICIENT * swipeAreaSize);
+        }
+        else
         {
-            touchActions.performVerticalSwipe(swipeAreaCenterY,
-                    swipeAreaCenterY - (elementTopCoordinateY - visibilityY), swipeArea, swipeDuration);
+            swipeAreaSize = swipeArea.getWidth();
+            elementCoordinate = SwipeDirection.LEFT == direction
+                    ? element.getLocation().getX() + element.getSize().getWidth() : element.getLocation().getX();
```

Comment:
I believe this logic should be completely symmetric to the logic from `if`-branch. But it's not for some reason

Reply:
fixed UP-DOWN logic